src/material-experimental/select/testing/shared.spec.ts

```diff
@@ -1,92 +1,64 @@
+import {OverlayContainer} from '@angular/cdk/overlay';
 import {HarnessLoader} from '@angular/cdk/testing';
 import {TestbedHarnessEnvironment} from '@angular/cdk/testing/testbed';
-import {Component, Type} from '@angular/core';
-import {ComponentFixture, TestBed, inject} from '@angular/core/testing';
-import {NoopAnimationsModule} from '@angular/platform-browser/animations';
-import {ReactiveFormsModule, FormControl, Validators} from '@angular/forms';
-import {OverlayContainer} from '@angular/cdk/overlay';
+import {Component} from '@angular/core';
+import {ComponentFixture, inject, TestBed} from '@angular/core/testing';
+import {FormControl, ReactiveFormsModule, Validators} from '@angular/forms';
 import {MatFormFieldModule} from '@angular/material/form-field';
 import {MatSelectModule} from '@angular/material/select';
-import {MatSelectModule as MatMdcSelectModule} from '../index';
+import {NoopAnimationsModule} from '@angular/platform-browser/animations';
 import {MatSelectHarness} from './select-harness';
-import {MatSelectHarness as MatMdcSelectHarness} from './mdc-select-harness';
-
-let fixture: ComponentFixture<SelectHarnessTest>;
-let loader: HarnessLoader;
-let harness: typeof MatSelectHarness;
-let overlayContainer: OverlayContainer;
-
-describe('MatSelectHarness', () => {
-  describe('non-MDC-based', () => {
-    beforeEach(async () => {
-      await prepareTests(MatSelectModule, SelectHarnessTest);
-      harness = MatSelectHarness;
-    });
-
-    runTests();
-  });
 
-  describe('MDC-based', () => {
-    beforeEach(async () => {
-      await prepareTests(MatMdcSelectModule, SelectHarnessTest);
-      // Public APIs are the same as MatSelectHarness, but cast
-      // is necessary because of different private fields.
-      harness = MatMdcSelectHarness as any;
-    });
-
-    // TODO: enable after MDC select is implemented
-    // runTests();
+/** Shared tests to run on both the original and MDC-based select. */
+export function runHarnessTests(
+    selectModule: typeof MatSelectModule, selectHarness: typeof MatSelectHarness) {
+  let fixture: ComponentFixture<SelectHarnessTest>;
+  let loader: HarnessLoader;
+  let overlayContainer: OverlayContainer;
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      imports: [
+        selectModule,
+        MatFormFieldModule,
```

Comment:
Wondering if we should pass such modules also through the `runHarnessTests` function. This will be a thing if we at some point have a MDC form-field. I guess it's not needed yet, so it's fine like that 😄

Reply:
yeah, we can switch it to take a list of modules or something when we get to it